### PR TITLE
cleaning up load balancer resources

### LINKED ISSUE #63026: GKE scalability jobs failing during Teardown
Our GKE scalability jobs (https://k8s-testgrid.appspot.com/sig-scalability-gke) are failing in the cluster teardown step (even though the e2e tests themselves passed). E.g failures:

- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-scale-correctness/10
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-scale-correctness/11

Seems like it's caused by some leaked network-related resources:

```
error deleting network: error during gcloud compute networks delete -q gke-scale-cluster --project=kubernetes-scale: exit status 1
```

/assign @krzysied 
Krzysiek - Could you PTAL into this when you find time? It'll be great to have our GKE suite back to green.

cc @kubernetes/sig-scalability-bugs @wojtek-t

### PR DESCRIPTION
**Pull request by krzysied**
**What this PR does / why we need it**:
Adds resources cleanup for some LoadBalancer tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
fixes #63026

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by shyamjvs** [comment]
/lgtm
/hold
(So @MrHohn can also TAL)

**Review by MrHohn — COMMENTED** [review]
Thanks for the fix!
cc @grayluck

File: test/e2e/network/service.go [hunk 1]
```diff
@@ -2023,6 +2023,9 @@ func execAffinityTestForLBService(f *framework.Framework, cs clientset.Interface
 	svc = jig.WaitForLoadBalancerOrFail(ns, serviceName, framework.LoadBalancerCreateTimeoutDefault)
 	jig.SanityCheckService(svc, v1.ServiceTypeLoadBalancer)
 	defer func() {
+		lb := cloudprovider.GetLoadBalancerName(svc)
+		framework.Logf("cleaning gce resource for %s", lb)
+		framework.CleanupServiceGCEResources(cs, lb, framework.TestContext.CloudConfig.Region, framework.TestContext.CloudConfig.Zone)
 		framework.StopServeHostnameService(cs, f.InternalClientset, f.ScalesGetter, ns, serviceName)
```
[the remark this answers is not in the document]
Comment (krzysied) [reply test/e2e/network/service.go]: Good catch!
I'll change order of function calls.

**Comment by shyamjvs** [comment]
/lgtm

Will let Zihong cancel the hold.

**Comment by MrHohn** [comment]
Thanks!

/lgtm
/hold cancel

**Comment by karataliu** [comment]
This change breaks non-gce providers:
```
Jun  6 16:36:18.011: INFO: Garbage collecting { ReplicationController} service pods took: 12.368365799s
Jun  6 16:36:18.459: INFO: cleaning gce resource for a04960e43696411e8a3d0000d3a11ab7
Jun  6 16:36:23.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:36:28.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:36:33.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:36:38.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:36:43.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:36:48.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:36:53.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:36:58.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:37:03.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:37:08.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:37:13.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:37:18.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:37:23.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
Jun  6 16:37:28.459: INFO: Still waiting for glbc to cleanup: failed to convert CloudConfig.Provider to GCECloud: <nil>
```

Should consider `CleanupServiceResources`

**Comment by MrHohn** [comment]
@karataliu Sorry, sent https://github.com/kubernetes/kubernetes/pull/64840 for fixing.

**Pull request merged**